### DISCUSSION TITLE
Question regarding `lru_cache`

### DISCUSSION THREAD

**Discussion by baseplate-admin** [hitblast/avro.py#39] [Q&A] [answered]
Hi, 

I have a question. Why do you use `lru_cache` extensively in this code base?

 Does `caching` a lookup really give that much of a performance improvement?

( dont take it as offensively, caching has a pitfall, i just want you to know that )

**Comment by hitblast** [comment]
@baseplate-admin Well I haven't really "extensively" used it. I've just implement a base amount of caching for the three main functions.

**Reply by baseplate-admin** [reply]
the problem is you haven't defined a `max_size` for those functions. Adding no `max_size` means the memory usage of `avro.py` in a web based (or any long running GUI) application will gradually increase to the point of OOM

**Reply by hitblast** [reply]
Oh, weren't they supposed to default to an integer value of `128` when there isn't a fixed number given?

**Reply by baseplate-admin** [reply]
I dont think that's the case, cause when i used it, in certain python versions,

They treat

`@lru_cache` as `@lru_cache(maxsize=None)`

Better be explicit about that

**Reply by hitblast** [reply]
@baseplate-admin If that's the case, a simple commit should resolve the issue.

**Reply by baseplate-admin** [reply]
> @baseplate-admin If that's the case, a simple commit should resolve the issue.

Yep. Please do remember this use case

**Reply by hitblast** [reply, marked as answer]
https://github.com/hitblast/avro.py/commit/3e794187077390cbe320cb9c73d414fe220f170e
